### DISCUSSION TITLE
Pagination facility is available only for Products not for Product Variants.

### DISCUSSION THREAD

**Discussion by ahmerdevopsinertia** [vendurehq/vendure#674] [Q&A]
Pagination facility is available only for Products not for Product Variants. I am using Vendure version 0.17.0

**Comment by michaelbromley** [comment]
Hi,
Can you be a bit more specific about the functionality you desire? Are you talking about paginating the variants of a single product, or do you want a new top-level query to return a paginated list of all variants?

**Reply by ahmerdevopsinertia** [reply]
Both will work, mean two new queries Pagination for all variants and Pagination for single product as well

**Reply by michaelbromley** [reply]
OK. The next minor release will include a top-level paginated variants query: https://github.com/vendure-ecommerce/vendure/commit/c2b4ed21d891b2f6165f14a4234d89b8bb53d605

**Reply by ahmerdevopsinertia** [reply]
Thanks let me know once available
